File: tourneyClient/src/models/entities/Characters/DrMario.ts
import { CharacterName } from "@/models/Enums/CharacterName";
import { Character } from "../Character"
import { CharacterId } from "@/models/Enums/CharacterId";
import { Archetype } from "@/models/Enums/Archetype";
import { FallSpeed } from "@/models/Enums/FallSpeed";
import { TierPlacement } from "@/models/Enums/TierPlacement";
import { WeightClass } from "@/models/Enums/WeightClass";

export const DrMario: Character =
{
    id: CharacterId.DrMario,
    characterName: CharacterName.DR_MARIO,
    archetype: Archetype.ALL_ROUNDER,
    fallSpeed: FallSpeed.FAST_FALLERS,
    weightClass: WeightClass.LIGHTWEIGHT,
    tierPlacement: TierPlacement.D

}

export default DrMario; 

File: tourneyAPI/Models/Entities/Characters/DrMario.cs
namespace Entities;

using Enums;

// Defines the competitive profile metadata for this playable character.
public class DrMario : Character
{
    // Initializes this character's default competitive attributes for matchmaking and tier logic.
    public DrMario()
    {
        Id = CharacterId.DrMario;
        characterName = CharacterName.DR_MARIO;
        archetype = Archetype.ALL_ROUNDER;
        fallSpeed = FallSpeed.FAST_FALLERS;
        weightClass = WeightClass.LIGHTWEIGHT;
        tierPlacement = TierPlacement.D;
    }

}